libtenzir/builtins/operators/load_tcp.cpp

```diff
@@ -0,0 +1,864 @@
+    TRY(set_close_on_exec(connection.socket->native_handle()));
+    if (args.tls) {
+      TENZIR_ASSERT(not connection.ssl_ctx);
+      connection.ssl_ctx.emplace(boost::asio::ssl::context::tls_server);
+      auto ec = boost::system::error_code{};
+      if (args.certfile) {
+        if (connection.ssl_ctx->use_certificate_chain_file(args.certfile->inner,
+                                                           ec)) {
+          return diagnostic::error("{}", ec.message())
+            .note("failed to load certificate chain file")
+            .primary(args.certfile->source)
+            .to_error();
+        }
+      }
+      if (args.keyfile) {
+        if (connection.ssl_ctx->use_private_key_file(
+              args.keyfile->inner, boost::asio::ssl::context::pem, ec)) {
+          return diagnostic::error("{}", ec.message())
+            .note("failed to load private key file")
+            .primary(args.certfile->source)
+            .to_error();
+        }
+      }
+      if (connection.ssl_ctx->set_verify_mode(boost::asio::ssl::verify_none,
+                                              ec)) {
+        return diagnostic::error("{}", ec.message())
+          .note("failed to disable peer certificate verification")
+          .primary(*args.tls)
+          .to_error();
+      }
+      TENZIR_ASSERT(not connection.tls_socket);
+      connection.tls_socket.emplace(*connection.socket, *connection.ssl_ctx);
+      if (connection.tls_socket->handshake(
+            boost::asio::ssl::stream<boost::asio::ip::tcp::socket>::server,
+            ec)) {
+        return diagnostic::error("{}", ec.message())
+          .note("failed to perform TLS handshake")
+          .primary(*args.tls)
+          .to_error();
+      }
+    }
+    // Set up and spawn the nested pipeline.
+    auto pipeline = args.pipeline->inner;
+    auto source = std::make_unique<load_tcp_source_operator>(
+      connection_actor{self}, connection.socket->native_handle());
+    pipeline.prepend(std::move(source));
+    auto sink = std::make_unique<load_tcp_sink_operator<Elements>>(
+      connection_manager_actor<Elements>{self});
+    pipeline.append(std::move(sink));
+    TENZIR_ASSERT(pipeline.is_closed());
+    TENZIR_ASSERT(not connection.pipeline_executor);
+    connection.pipeline_executor = self->template spawn<caf::monitored>(
+      pipeline_executor, std::move(pipeline), receiver_actor<diagnostic>{self},
+      metrics_receiver_actor{self}, node, has_terminal, is_hidden);
+    self->request(connection.pipeline_executor, caf::infinite, atom::start_v)
+      .then(
+        [this, handle = connection.socket->native_handle()]() {
+          // Start the async read loop for this connection.
+          auto connection = connections.find(handle);
+          TENZIR_ASSERT(connection != connections.end());
+          connection->second.async_read(self);
+        },
+        [](const caf::error& err) {
+          // FIXME: Emit a diagnostic
```

Comment:
We need to decide here whether failing to start the nested pipeline is an error or a warning.